s3/src/test/scala/akka/stream/alpakka/s3/impl/DiskBufferSpec.scala

```diff
@@ -84,4 +84,12 @@ class DiskBufferSpec(_system: ActorSystem)
     }
 
   }
+
+  it should "not emit rest chunk if rest chunk is empty" in {
```

Comment:
Does this test validate that the extra element is only emitted if the `MemoryBuffer` is used for chunking?